src/grpc.rs

```diff
@@ -550,6 +601,33 @@ impl ClientSettings {
                             }
                         }
 
+                        "defaultdeadline" => {
+                            let value = values.as_slice()[1];
+
+                            if let Ok(int) = value.parse::<i64>() {
+                                if int == -1 {
+                                    result.default_deadline = Some(Duration::from_millis(u64::MAX));
+                                    continue;
+                                }
+
+                                if int < -1 {
+                                    error!("Invalid defaultDeadline of {}. Please provide a positive integer, or -1 to disable", int);
```

Comment:
**SUGGESTION:** I might be missing, but I don't see a negative test case for this error.